### PR TITLE
Add const lvalue ref to container parameters

### PR DESCRIPTION
**Pull request by Muller-Castro**
Adds const lvalue reference to the core container types specified in the [documentation](https://docs.godotengine.org/en/stable/development/cpp/core_types.html) that are declared in function parameters.

`core/*`: https://github.com/godotengine/godot/pull/86966
`editor/*`: https://github.com/godotengine/godot/pull/88368
`modules/*`: https://github.com/godotengine/godot/pull/88915
`platform/*`: https://github.com/godotengine/godot/pull/88971
`servers/*`: https://github.com/godotengine/godot/pull/88972
`scene/*`: https://github.com/godotengine/godot/pull/88974
others: https://github.com/godotengine/godot/pull/88975

### CODE REVIEW

**Comment by reduz** [comment]
This would probably be good to rebase, its a nice clean up.

**Comment by YuriSizov** [comment]
As mentioned above, needs a rebase, but is a welcome change.

**Comment by Muller-Castro** [comment]
Hello! Sorry for the delay 😓I had to step away from programming for a couple of years.

I applied the fix to the other classes that are on that doc page using a python script.

Please let me know if you have any more classes that need this fix or if there is somewhere in the documentation where I can look for more classes that are not cheap when copying.

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot/pull/75819

Needs to see if that one is necessary with these changes

**Review by AThousandShips — DISMISSED** [review]
These all look good to me

I'd personally add `String` and `Vector` to methods already touched by this, `StringName` is as cheap to copy as those are, but it's a minor improvements for them given that we avoid doing synchronization

**Comment by Muller-Castro** [comment]
Fixed for `String`, now I'm fixing `Vector` and its typedefs defined [here](https://github.com/godotengine/godot/blob/master/core/variant/variant.h#L68). Do you think it's a good idea to do the same for `Array`, `Dictionary` and `Variant`?

**Comment by AThousandShips** [comment]
I'd say it's generally worth it, minor detail but there's COW semantics and refcounting that's not trivial, and it's free to pass by reference

**Comment by AThousandShips** [comment]
If you need help with resetting this please don't hesitate to ask, I'd prefer to merge this with the original set of changes and review the remaining later as it's much larger in scope 🙂

**Comment by AThousandShips** [comment]
You'd create a new branch from this one, and then reset this one, you can then create a new PR with the larger changes

The changes needs to be reviewed to prevent any issues, it can have messy side effects that can be hard to predict, your current changes aren't trivial and involves copying data in some cases, that'd require a more thorough review

Either this can be merged now with the original changes, or reviewed later with a larger scope, I'm not comfortable merging it without further review in its current state 🙂

File: core/core_bind.cpp [hunk 1]
```diff
@@ -52,11 +52,13 @@ Error ResourceLoader::load_threaded_request(const String &p_path, const String &
 	return ::ResourceLoader::load_threaded_request(p_path, p_type_hint, p_use_sub_threads, ResourceFormatLoader::CacheMode(p_cache_mode));
 }
 
-ResourceLoader::ThreadLoadStatus ResourceLoader::load_threaded_get_status(const String &p_path, Array r_progress) {
+ResourceLoader::ThreadLoadStatus ResourceLoader::load_threaded_get_status(const String &p_path, const Array &r_progress) {
```
Comment (AThousandShips) [inline core/core_bind.cpp:55]: This, for example, isn't a valid change, the argument is mutated, and should instead be:
```suggestion
ResourceLoader::ThreadLoadStatus ResourceLoader::load_threaded_get_status(const String &p_path, Array &r_progress) {
```
So important to get this in a separate thing for cases that aren't obvious or simple

**Review by AThousandShips — APPROVED** [review]

**Comment by Muller-Castro** [comment]
I rollbacked using this command: `git reset --hard 9062dee` and then rebased to see if there was any conflict.

Should I pull another request with the remaining changes?

**Comment by AThousandShips** [comment]
They should really be split into a few different PRs, separated by part of the engine, like `core`, `scene`, `modules`, etc., over 400 changes aren't easy or manageable to review

**Comment by akien-mga** [comment]
Thanks for your patience and great work!

**Pull request merged**